### ISSUE TITLE
Micromap crashes Rack 2.5.0 arm64

### ISSUE THREAD

**Issue by fractalgee** [stoermelder/vcvrack-packone#385]
Hi Ben,

looks like the 2.5.0 Rack update broke MicroMap, It is now reliebly crashing Rack, which renders a whole bunch of my patches unusable as i use it all over the place. worked peachy with Rack 2.4.1 and prior.

Crash is always same:

```
[8.928 fatal adapters/standalone.cpp:49 fatalSignalHandler] Fatal signal 6. Stack trace:
23: Rack(fatalSignalHandler(int)+40)
22: libsystem_platform.dylib(_sigtramp+56)
21: libsystem_pthread.dylib(pthread_kill+288)
20: libsystem_c.dylib(abort+180)
19: libsystem_c.dylib(err+0)
18: libRack.dylib(rack::engine::Engine::toJson() (.cold.1)+0)
17: libRack.dylib(rack::engine::Engine::updateParamHandle_NoLock(rack::engine::ParamHandle*, long long, int, bool)+80)
16: libRack.dylib(rack::engine::Engine::updateParamHandle(rack::engine::ParamHandle*, long long, int, bool)+92)
15: plugin.dylib(StoermelderPackOne::MapModuleBase<1>::clearMap(int)+80)
14: libRack.dylib(rack::widget::Widget::step()+64)
13: plugin.dylib(StoermelderPackOne::ThemedModuleWidget<StoermelderPackOne::CVMapMicro::CVMapMicroModule, rack::app::ModuleWidget>::step()+420)
12: libRack.dylib(rack::widget::Widget::step()+64)
11: libRack.dylib(rack::widget::Widget::step()+64)
10: libRack.dylib(rack::widget::Widget::step()+64)
9: libRack.dylib(rack::widget::Widget::step()+64)
8: libRack.dylib(rack::widget::Widget::step()+64)
7: libRack.dylib(rack::ui::ScrollWidget::step()+24)
6: libRack.dylib(rack::app::RackScrollWidget::step()+560)
5: libRack.dylib(rack::widget::Widget::step()+64)
4: libRack.dylib(rack::app::Scene::step()+432)
3: libRack.dylib(rack::window::Window::step()+1540)
2: libRack.dylib(rack::window::Window::run()+40)
1: Rack(main+4080)
0: dyld(start+2360)
```

Hope this can be fixed quickly as otherwise I am rolling back to 2.4.1 due to so many patches now crashing Rack.

P1 is version 2.0.beta3 as per plugin json. arm64 MacOS 14.4.2.  I also tried building v2-dev branch here but that gets me 2.0.0 and still crashes.

**Comment by stoermelder** [comment]
Thanks for reporting. I can reproduce these crashes with VCV's own MIDI-MAP, so this is caused by the update itself. I will report an issue.

**Comment by stoermelder** [comment]
Steps to reproduce the issue:
1. Start with the default template patch
2. Add MIDI-MAP module
3. Map any parameter in the patch, it doesn’t matter which one, a MIDI assignment is not needed
4. Close Rack
5. Start Rack, the previous patch is autoloaded
6. Remove MIDI-MAP module

**Comment by fractalgee** [comment]
just verified, will report also to support

**Comment by stoermelder** [comment]
This issue has been fixed by VCV with upcoming Rack v2.5.1
https://github.com/VCVRack/Rack/blob/v2/CHANGELOG.md#251-2024-04-16

**Comment by fractalgee** [comment]
I can confirm that 2.5.1 no longer crashes when I load MIDI mapped (yours, patchmaster and VCV's) patches, or they are in the autostart.
